monitoring/uss_qualifier/resources/resource.py

```diff
@@ -35,85 +38,75 @@ def is_type(self, resource_type: str) -> bool:
 ResourceType = TypeVar("ResourceType", bound=Resource)
 
 
-class ResourceDeclaration(ImplicitDict):
-    resource_type: str
-    """Type of resource, expressed as a Python class name qualified relative to this `resources` module"""
-
-    dependencies: Dict[str, str] = {}
-    """Mapping of dependency parameter (additional argument to concrete resource constructor) to `name` of resource to use"""
-
-    specification: dict = {}
-    """Specification of resource; format is the SpecificationType that corresponds to the `resource_type`"""
-
-    def make_resource(self, resource_pool: Dict[str, Resource]) -> Resource:
-        inspection.import_submodules(resources_module)
-        resource_type = inspection.get_module_object_by_name(
-            resources_module, self.resource_type
-        )
-        if not issubclass(resource_type, Resource):
-            raise NotImplementedError(
-                "Resource type {} is not a subclass of the Resource base class".format(
-                    resource_type.__name__
-                )
-            )
+def create_resources(
+    resource_declarations: Dict[ResourceID, ResourceDeclaration]
+) -> Dict[ResourceID, ResourceType]:
+    resource_pool: Dict[ResourceID, ResourceType] = {}
 
-        constructor_signature = inspect.signature(resource_type.__init__)
-        specification_type = None
-        constructor_args = {}
-        for arg_name, arg in constructor_signature.parameters.items():
-            if arg_name == "self":
-                continue
-            if arg_name == "specification":
-                specification_type = arg.annotation
+    resources_created = 1
+    while resources_created > 0:
```

Comment:
Do we still need this loop ? resources_created doesn't seem to be called and I don't see control flow statements for the while

Reply:
Heh, multiple mistakes apparently added up to tests passing :)  Yes, the idea here is to loop through all the resources and only create ones that don't have any unmet dependencies, then repeat that loop until no resources were created (we're finished or stuck).  We do need this condition when the logic is corrected (which is done now, hopefully).